### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by glebm**
@pcercuei Can you push the missing platforms/pc directory?

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,6 +19,21 @@ if (BIND_CONSOLE)
 	add_compile_definitions(BIND_CONSOLE)
 endif (BIND_CONSOLE)
 
+option(WINDOWED_MODE "Run windowed" OFF)
+if (WINDOWED_MODE)
+	add_compile_definitions(G2X_BUILD_OPTION_WINDOWED_MODE)
+endif ()
+
+option(SCREEN_WIDTH "Screen / window width" OFF)
```
Comment (pcercuei) [inline CMakeLists.txt]: ```option``` is just for ON/OFF values, so you cannot use it to pass integers. It won't work in the CMake GUI.
What you had before (with the default "max", which could be changed by "best") did work better.

File: .gitignore [hunk 2]
```diff
@@ -16,3 +16,4 @@
 #	src/.deps/
 #	src/Makefile
 #	src/Makefile.in
+/build*/
```
Comment (pcercuei) [inline .gitignore:19]: Just ```build``` is fine no? Why the trailing slash?

**Review by pcercuei — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (glebm) [reply .gitignore:19]: I usually use separate build directories for different targets:

```bash
cmake -S. -Bbuild # pc
cmake -S. -Bbuild-rg350 -DCMAKE_TOOLCHAIN_FILE=... # rg350
```

This preserves the build cache for each target and makes my command history more useful

**Comment by pcercuei** [comment]
About the platforms/pc directory it's just the gcw0 one on my side. So there's nothing really specific here.

With that said, I think GMenu2X should not have any notion of platform, and that the platform-specific files should be installed by the Buildroot.

File: .gitignore (hunk 2 above)
Comment (glebm) [reply .gitignore:19]: The trailing slash is not needed. I like it for clarity when specifying directories though.

File: CMakeLists.txt (hunk 1 above)
Comment (glebm) [reply CMakeLists.txt]: Thanks, fixed. Decided on empty string as default this time.

**Comment by glebm** [comment]
Ping. Is there anything blocking this from being merged?

File: CMakeLists.txt [hunk 3]
```diff
@@ -19,6 +19,21 @@ if (BIND_CONSOLE)
 	add_compile_definitions(BIND_CONSOLE)
 endif (BIND_CONSOLE)
 
+option(WINDOWED_MODE "Run windowed" OFF)
+if (WINDOWED_MODE)
+	add_compile_definitions(G2X_BUILD_OPTION_WINDOWED_MODE)
+endif ()
+
+set(SCREEN_WIDTH "" CACHE STRING "Screen / window width")
```
Comment (pcercuei) [inline CMakeLists.txt]: Maybe change it to: ```Screen / window width (empty: max available)```

**Comment by pcercuei** [comment]
Apart from the comment above, I don't want to merge the .clang-format right now. I think it's a good idea but we should discuss the style first.

File: CMakeLists.txt (hunk 3 above)
Comment (glebm) [reply CMakeLists.txt]: Done

**Comment by glebm** [comment]
Removed .clang-format (will send as a separate PR for discussion)

**Comment by pcercuei** [comment]
Merged, thanks.

**Pull request merged**